FP-1572: Solve Unequal Column Overflow Bug, But Retain 60/40 Layout

wesleyboar opened the pull request:
## Overview

- Revert https://github.com/TACC/Core-Portal/pull/626 (which changed layout).
- Fix overflow bug by fixing what broke the overflow bug (which retains layout).

## Related

* [FP-1572](https://jira.tacc.utexas.edu/browse/FP-1572)
* expected to be used by https://github.com/TACC/Core-Portal/pull/688

## Changes

- restore removed classes for Dashboard and ManageAccount
- fix mangled global class that was originally used to fix overflow bug

## Testing

1. Verify Dashboard has 60/40 column layout.
1. Verify Dashboard has no [horizontal overflow](https://jira.tacc.utexas.edu/browse/FP-1572).
1. Verify Manage Account has 60/40 column layout.
1. Verify Manage Account has no [horizontal overflow](https://jira.tacc.utexas.edu/browse/FP-1572).
2. (extra) Verify Dashboard and Manage Account at different layouts behave as they had before\*.

<sub>\* Vague on purpose. Sorry. I haven't had time to find the PR that illustrates the full responsive design.</sub>

## UI

| Manage Account (Same Panels) | Dashboard (w/ Custom App) |
| - | - |
| ![Manage Account (No Change)](https://user-images.githubusercontent.com/62723358/186791961-82bd19f6-3ee8-42fe-b4b6-652607a924c4.png) | ![Dashboard With Custom App](https://user-images.githubusercontent.com/62723358/186791964-bcdc77db-823e-4850-b226-7d08571ad8e7.png) |

## Notes

This does not affect other layouts than Dashboard and ManageAccount, because no layouts were using the layouts I've given back to them.

### Improvements that Could be Made

[FP-1603](https://jira.tacc.utexas.edu/browse/FP-1603): Replace unequal columns with CSS multi-column layout with CSS grid. (I should have used CSS Grid when I originally created this layout, but I was not competent enough with it yet.)

https://github.com/TACC/Core-Portal/pull/690#discussion_r956175886: Use something like `count--${panelCount}` to manage small width generically rather than specifically.

rstijerina reviewed (commented):
Nice, this solves the biggest issue with the `twoColumnUnequal` layout. There is still some overflow over the `<main>` element margin-right. It's not obvious since the System Monitor has bespoke css to make the width short, but it's clear when reordering that component in the dashboard, or by adding the CustomDashboardSection:
![Screen Shot 2022-08-25 at 4 00 22 PM](https://user-images.githubusercontent.com/20326896/186768299-fb8d0008-7dd9-4cd1-8633-5472f8c067df.png)
![Screen Shot 2022-08-25 at 3 59 54 PM](https://user-images.githubusercontent.com/20326896/186768304-21186522-32c3-4e7c-88d6-753ee1f34be0.png)

wesleyboar commented:
Good eye. Thanks. I'll check.

I am looking at the changes to https://github.com/TACC/Core-Portal/pull/688 first.

wesleyboar commented:
> I feel like this could be overcome by applying the margin-left and margin-right of a section component to the top `root` level of a section component, from the `<main>` element where it is currently

I had felt (when originally building sections) and now feel (seeing your comment) that also.

**But** margin is ineffectual **and** padding moves everything it should… except the horizontal lines of the table (because the width of the second column is unchanged… even when I remove the `.sysmon-panel` width).

<sub>P.S. CSS Multi-Column Layout was the wrong tool for the job. This ought to use CSS Grid. I am _still_ sorry I laid out columns this way.</sub>

wesleyboar commented:
@rstijerina,

I added a classname `.custom-panel` to the custom panel, so I could constrain its width the same way I did for `.sysmon-panel`. To add the classname, I added a `className` prop to `CustomDashboardSection`. Please review 2aebef6.

rstijerina reviewed (approved):
LGTM, thanks for the work on this

nathanfranklin reviewed (approved):
the layout changes look good to me 👍 .  but note that client tests are still failing.  I see it uses the being reviewed PR https://github.com/TACC/Core-Portal/pull/688.  is this PR going to wait for that one to be merged to main before this is merged to main?
